README.md

```diff
@@ -234,31 +253,34 @@ const items = new BehaviorSubject<TodoItem[]>([
 ]);
 ```
 
-We can then define a function to create a component from an individual item observable:
-```typescript
-const Item = (item: Observable<TodoItem>) => Div(
-  item.pipe(
+We can then define a component taking an individual item observable:
+
+```tsx
+const Item: FC<{ item: Observable<TodoItem> }> = ({ item }) => <div>
+  {item.pipe(
     map(({ name, done }) => `${name} is ${done ? '' : 'not'} done!`),
-  ),
-);
+  )}
+</div>;
 ```
 
-Using the `mapToComponents` operator function, we can map the item array into an array of `Item` components. Here the first argument is a function taking an item and returning its unique id (similar to the 'key' prop in React).
-```typescript
+Using the `mapToComponents` operator function, we can map the item array into an array of `Item` components. Here the second argument is either a function taking the item and returning its unique id (similar to the 'key' prop in React) or a property name on the item where its unique id can be found. If this argument is omitted then the item's index in the source array will be used as its id.
+
+```tsx
 const ItemComponents = items.pipe(
-  mapToComponents(item => item.name, Item),
-);
+    mapToComponents(item => <Item item={item} />, 'name'),
+  );
 ```
 
 The resulting component array observable can be passed directly as a component child:
 ```typescript
-const ComponentArraysExample = Div(ItemComponents);
+const ComponentArraysExample = <div>{ItemComponents}</div>;
 ```
 
 If our `items` subject were to then emit a new array, this would be immediately be reflected by our `Item` components in the DOM. Any items with matching ids from the previous emission will reuse the existing DOM elements. 
```

Comment:
```suggestion
If our `items` subject were to then emit a new array, this would immediately be reflected by our `Item` components in the DOM. Any items with matching ids from the previous emission will reuse the existing DOM elements. 
```